src/server/stream_family.cc

```diff
@@ -1364,6 +1502,124 @@ void StreamFamily::XAdd(CmdArgList args, ConnectionContext* cntx) {
   return (*cntx)->SendError(add_result.status());
 }
 
+void StreamFamily::XClaim(CmdArgList args, ConnectionContext* cntx) {
+  ClaimOpts opts;
+  string_view key = ArgS(args, 0);
+  opts.group = ArgS(args, 1);
+  opts.consumer = ArgS(args, 2);
+  if (!absl::SimpleAtoi(ArgS(args, 3), &opts.min_idle_time)) {
+    return (*cntx)->SendError(kSyntaxErr);
+  }
+  // Ignore negative min-idle-time
+  opts.min_idle_time = std::max(opts.min_idle_time, static_cast<int64>(0));
+  args.remove_prefix(4);
+
+  auto GetXclaimIds = [&]() {
+    // get the ids
+    size_t i;
+    absl::InlinedVector<streamID, 8> ids;
+    for (i = 0; i < args.size(); ++i) {
+      ParsedStreamId parsed_id;
+      string_view str_id = ArgS(args, i);
+      if (!ParseID(str_id, true, 0, &parsed_id)) {
+        if (i > 0) {
+          break;
+        }
+        return ids;
+      }
+      ids.push_back(parsed_id.val);
+    }
+    args.remove_prefix(i);
+    return ids;
+  };
+
+  auto ids = GetXclaimIds();
+  if (!ids.size()) {
+    // No ids given.
+    return (*cntx)->SendError(kInvalidStreamId, kSyntaxErrType);
+  }
+
+  // parse the options
+  auto ParseXclaimOptions = [&]() {
+    size_t i = 0;
+    for (i = 0; i < args.size(); ++i) {
+      ToUpper(&args[i]);
+      string_view arg = ArgS(args, i);
+      bool remaining_args = args.size() - i - 1 > 0;
+
+      if (remaining_args) {
+        if (arg == "IDLE") {
+          arg = ArgS(args, ++i);
+          if (!absl::SimpleAtoi(arg, &opts.deliverytime)) {
+            return (*cntx)->SendError(kInvalidIntErr);
+          }
+          continue;
+        } else if (arg == "TIME") {
+          arg = ArgS(args, ++i);
+          if (!absl::SimpleAtoi(arg, &opts.deliverytime)) {
+            return (*cntx)->SendError(kInvalidIntErr);
+          }
+          continue;
+        } else if (arg == "RETRYCOUNT") {
+          arg = ArgS(args, ++i);
+          if (!absl::SimpleAtoi(arg, &opts.retry)) {
+            return (*cntx)->SendError(kInvalidIntErr);
+          }
+          continue;
+        } else if (arg == "LASTID") {
+          arg = ArgS(args, ++i);
+          // TODO: implement lastID
+          if (!ParseID(arg, true, 0, &opts.lastid)) {
+            return (*cntx)->SendError(kInvalidStreamId, kSyntaxErrType);
+          }
+          continue;
+        }
+      }
+      if (arg == "FORCE") {
+        opts.flags |= kClaimForce;
+      } else if (arg == "JUSTID") {
+        opts.flags |= kClaimJustID;
+      } else {
+        return (*cntx)->SendError("Unknown argument given for XCLAIM command", kSyntaxErr);
+      }
+    }
+  };
+
+  ParseXclaimOptions();
+  if (auto now = GetCurrentTimeMs();
+      opts.deliverytime < 0 || static_cast<uint64_t>(opts.deliverytime) > now)
+    opts.deliverytime = now;
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return OpClaim(t->GetOpArgs(shard), key, opts, absl::Span{ids.data(), ids.size()});
+  };
+
```

Comment:
I'd move this empty line to be after the below `ScheduleSingleHopT()`, as the `cb` is for that